tracing-subscriber/tests/subscriber_filters/option.rs

```diff
@@ -0,0 +1,40 @@
+use super::*;
+use tracing_subscriber::{filter, prelude::*, Subscribe};
+
+fn filter<S>() -> filter::DynFilterFn<S> {
+    // Use dynamic filter fn to disable interest caching and max-level hints,
+    // allowing us to put all of these tests in the same file.
+    filter::dynamic_filter_fn(|_, _| false)
+}
+
+#[test]
+fn option_some() {
+    let (subscribe, handle) = subscriber::mock().only().run_with_handle();
+    let subscribe = Box::new(subscribe.with_filter(Some(filter())));
```

Comment:
nit: i'm not sure if `Box` is necessary, but if it is, then that's fine.

Reply:
I copied `boxed.rs` as a starting point. I can unbox.